broadcast/internal/arc/arc_submit_tx.go

```diff
@@ -144,20 +154,45 @@ func createSubmitBatchTxsBody(txs []*broadcast.Transaction) ([]byte, error) {
 	return data, nil
 }
 
-func appendSubmitTxHeaders(pld *httpclient.HTTPRequest, tx *broadcast.Transaction) {
-	if tx.MerkleProof {
+func WithCallback(callbackURL string, callbackToken ...string) broadcast.TransactionOptFunc {
+	return func(o *broadcast.TransactionOpts) {
+		o.CallbackToken = callbackURL
+		if len(callbackToken) > 0 {
+			o.CallbackToken = callbackToken[0]
+		}
+	}
+}
+
+func WithMerkleProof() broadcast.TransactionOptFunc {
+	return func(o *broadcast.TransactionOpts) {
+		o.MerkleProof = true
+	}
+}
+
+func WithWaitForStatus(status broadcast.TxStatus) broadcast.TransactionOptFunc {
+	return func(o *broadcast.TransactionOpts) {
+		o.WaitForStatus = status
+	}
+}
+
+func appendSubmitTxHeaders(pld *httpclient.HTTPRequest, opts *broadcast.TransactionOpts) {
+	if opts == nil {
+		return
+	}
+
+	if opts.MerkleProof {
 		pld.AddHeader("X-MerkleProof", "true")
 	}
 
-	if tx.CallBackURL != "" {
-		pld.AddHeader("X-CallbackUrl", tx.CallBackURL)
+	if opts.CallbackURL != "" {
+		pld.AddHeader("X-CallbackUrl", opts.CallbackURL)
 	}
 
-	if tx.CallBackToken != "" {
-		pld.AddHeader("X-CallbackToken", tx.CallBackToken)
+	if opts.CallbackToken != "" {
+		pld.AddHeader("X-CallbackToken", opts.CallbackToken)
 	}
 
-	if statusCode, ok := broadcast.MapTxStatusToInt(tx.WaitForStatus); ok {
+	if statusCode, ok := broadcast.MapTxStatusToInt(opts.WaitForStatus); ok {
 		pld.AddHeader("X-WaitForStatus", strconv.Itoa(statusCode))
 	}
 }
```

Comment:
## minor
I would rather want to see it as those `With***` will append headers by themselves.
I will try to present that in the following pseudo code:
```go
func submitTransaction(ctx context.Context, arc *ArcClient, tx *broadcast.Transaction, opts ...broadcast.TransactionOptFunc) (*broadcast.SubmitTxResponse, error) {
       // (...)
       // instead of this => appendSubmitTxHeaders(&pld, opts), there will be following: 
      	for _, opt := range opts {
		opt(&pld)
	}
     
      //(...)
}


func WithCallback(callbackURL string, callbackToken ...string) broadcast.TransactionOptFunc {
	return func(o *broadcast.TransactionOpts) {
		pld.AddHeader("X-CallbackUrl", callbackURL)

	        if len(callbackToken) > 0 {
		    pld.AddHeader("X-CallbackToken", callbackToken[])
	        }
	}
}

// ... rest of With methods
```